hooks/contracts/useClaimableTokens.ts

```diff
@@ -0,0 +1,37 @@
+import { useAtomValue, useSetAtom } from 'jotai'
+import type { BigNumber } from 'ethers'
+import { formatUnits } from 'ethers/lib/utils'
+import { useContractRead } from 'wagmi'
+
+import {
+  claimableTokensAtom,
+  liquidityGaugeAddressAtom,
+  stakingContractAddressAtom,
+} from 'states/staking'
+import { networkChainId } from 'utils/network'
+import { findAbiFromLiquidityGauge } from 'utils/wagmi'
+
+const FN = 'claimable_tokens'
+const ABI = findAbiFromLiquidityGauge(FN)
+
+export function useClaimableTokens() {
+  const liquidityGaugeAddress = useAtomValue(liquidityGaugeAddressAtom)
+  const stakingAddress = useAtomValue(stakingContractAddressAtom)
+  const setClaimableTokens = useSetAtom(claimableTokensAtom)
+
+  useContractRead({
```

Comment:
Consider using the `data` property directly instead of `claimableTokensAtom`. This review likely applies to other hooks in `hooks/contracts`.

If using multiple `useContractRead` causes other issues such as duplicate updates I think using Jotai as a proxy is a great idea. But if it isn't the case, it is worth reconsidering.

Reply:
Reading wagmi's source code suggests that they are stored in TanStack Query's cache meaning it would be safe to use them around.

Reply:
@BasixKOR I've tested what you suggested and found out that `useContractRead(s)` are called on every hook calls when `watch: true` option is present. So I changed the code to get `data` directly from `useContractRead(s)` when its `cacheTime` is set to `Infinity`. 👍